Sources/Amplitude/Amplitude.m

```diff
@@ -1020,26 +1020,22 @@ - (void)makeEventUploadPostRequest:(NSString *)url events:(NSString *)events num
         BOOL uploadSuccessful = NO;
         NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
         if (response != nil) {
+            NSString *result = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
             if ([httpResponse statusCode] == 200) {
-                NSString *result = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
-                if ([result isEqualToString:@"success"]) {
-                    // success, remove existing events from dictionary
-                    uploadSuccessful = YES;
-                    if (maxEventId >= 0) {
-                        (void) [self.dbHelper removeEvents:maxEventId];
-                    }
-                    if (maxIdentifyId >= 0) {
-                        (void) [self.dbHelper removeIdentifys:maxIdentifyId];
-                    }
-                } else if ([result isEqualToString:@"invalid_api_key"]) {
-                    AMPLITUDE_ERROR(@"ERROR: Invalid API Key, make sure your API key is correct in initializeApiKey:");
-                } else if ([result isEqualToString:@"bad_checksum"]) {
-                    AMPLITUDE_ERROR(@"ERROR: Bad checksum, post request was mangled in transit, will attempt to reupload later");
-                } else if ([result isEqualToString:@"request_db_write_failed"]) {
-                    AMPLITUDE_ERROR(@"ERROR: Couldn't write to request database on server, will attempt to reupload later");
-                } else {
-                    AMPLITUDE_ERROR(@"ERROR: %@, will attempt to reupload later", result);
+                // success, remove existing events from dictionary
+                uploadSuccessful = YES;
+                if (maxEventId >= 0) {
+                    (void) [self.dbHelper removeEvents:maxEventId];
                 }
+                if (maxIdentifyId >= 0) {
+                    (void) [self.dbHelper removeIdentifys:maxIdentifyId];
+                }
+            } else if ([result isEqualToString:@"invalid_api_key"]) {
```

Comment:
What's the response code for this case?

Reply:
Thanks for reviewing! This case I know it is 400 for `invalid_api_key`. Not sure about the `bad_checksum` and `request_db_write_failed` cases. I am following the similar logic as [Android](https://github.com/amplitude/Amplitude-Android/blob/1e16e9e05c64309e6d23c10647fefd80353f607f/src/main/java/com/amplitude/api/AmplitudeClient.java#L2296-L2304), seems there is no official doc for this endpoint. Any suggestion where we can get the answer?

Reply:
Checking with pipeline team, let me know if you have other suggestions.

Reply:
That's the right direction. Also can ask them for where the code lives.

Reply:
- Updated the `400` code for `invalid_api_key` and `bad_checksum`.
- PL team is not aware of the `request_db_write_failed` error though, they are assuming it is `5xx`, I put it `500` here, searching in the codebase doesn't give me answer on this either, think it might be legacy code, but no harm to leave it here.

@haoliu-amp , let me know your thoughts, thanks!

Reply:
That's a different endpoint other than HTTP API v2. You can ask Curtis.